### PR TITLE
Refactor actions and support caching based on the many potential configuration files Jekyll supports

### PR DESCRIPTION
**Pull request by yoannchaudet**
More refactoring. Actions now share a shell script (symlink).

The caching action supports the three kind of configuration files a Jekyll project may have.

### CODE REVIEW

File: compute-cache-key/run.sh [hunk 1]
```diff
@@ -1,56 +1,30 @@
...
-  cache_path=".pages/".$(cat /proc/sys/kernel/random/uuid)
-done
-
-# Output the temporay path to restore cache.
-echo "::set-env name=PAGES_CACHE_PATH::${cache_path}"
+# Compute a hash key and expose it in an environment variable
+aggregated_hash="
+  $(hash_folder "${PAGES_ROOT_FOLDER}"/_plugins)
+  $(hash_file "$(first_file "${PAGES_ROOT_FOLDER}"/_config.yml "${PAGES_ROOT_FOLDER}"/_config.toml "${PAGES_ROOT_FOLDER}"/_config.yaml)")
+  $(hash_file "$(first_file "${PAGES_ROOT_FOLDER}"/Gemfile.lock "${PAGES_ROOT_FOLDER}"/Gemfile.lock)")
```
Comment (yoannchaudet) [inline compute-cache-key/run.sh]: I need to bring back the warning. I forgot it 🙈

Comment (YiMysty) [inline compute-cache-key/run.sh]: remove the second `.lock` :)

File: build-jekyll/Dockerfile [hunk 2]
```diff
@@ -2,8 +2,8 @@
 # Source: https://github.com/envygeeks/jekyll-docker
 FROM jekyll/builder:latest
 
-# Copy run script and make it executable
-COPY run.sh /
+# Copy scripts
```
Comment (YiMysty) [inline build-jekyll/Dockerfile]: currently `build-jekyll` function use `#!/usr/bin/env bash` need to add 
```
# Install bash
COPY run.sh /	RUN apk add --no-cache bash
```

Comment (yoannchaudet) [reply build-jekyll/Dockerfile]: Bash is available in their image.

**Pull request merged**